Tradução: quarto.qmd (#81)
cienciadedatos/pt-r4ds: quarto.qmd

source("_common.R")
```

## Introdução

O Quarto fornece uma estrutura (*framework*) unificada para autoria em ciência de dados, combinando seu código, seus resultados e seu texto.
Documentos Quarto são plenamente reproduzíveis e suportam muitos formatos de saída como PDF, arquivos Word, apresentações e mais.

Arquivos Quarto foram projetados para serem usados de três maneiras:

> 1.  Para se comunicar com pessoas tomadoras de decisões, que querem focar nas conclusões e não no código por trás da análise.

Reviewer: ```suggestion
1.  Para se comunicar com pessoas tomadoras de decisão, que querem focar nas conclusões e não no código por trás da análise.
```

Na verdade não sei se fica melhor, mas achei estranho "decisões"

2.  Para colaborar com outras pessoas cientistas de dados (incluindo você no futuro!) que estejam interessadas tanto em suas conclusões quanto em como você as alcançou (isto é, o código).

3.  Como um ambiente em que se *faz* ciência de dados, como um caderno (*notebook*) de laboratório moderno em que se pode capturar não apenas o que você fez, como também o que você estava pensando.

Quarto é uma feerramenta de linha de comando, não é um pacote do R.
Isso significa que ajuda, via de regra, não está disponível via `?`.
Em vez disso, enquanto você trabalhar neste capítulo e usar o Quarto no futuro, você deve procurar a [documentação para Quarto](https://quarto.org).

Se você for uma pessoa que usa R Markdown, você deve estar pensando "Quarto parece muito com R Markdown".
Isso não está errado!
Quarto une a funcionalidade de muitos pacotes do ecosistema do R Markdown (rmarkdown, bookdown, distill, xaringan, etc.) em um sistema único e consistente, mas também extende o ecosistema com apoio nativo para múltiplas linguagens de programação como Python e Julia, além do próprio R.

> De certa forma, Quarto reflete tudo que foi aprendido na expansão e suporte ao ecosistema do R Markdown em uma década.

Reviewer: ```suggestion
De certa forma, o Quarto reflete tudo que foi aprendido na expansão e no suporte ao ecossistema do R Markdown em uma década.
```
acho que em Português fica melhor " o Quarto"... a não ser que seja o padrão que foi escolhido na tradução (de nao usar o artigo)... se não acho que soa entranho em português, sem o artigo. 

E a principio acho q temos que seguir o paralelismo na regência do verbo  qual tem mais de um objeto

### Pré-requisitos

> Você precisa da interface de linha de comando Quarto (Quarto CLI -- *command line interface*), mas não é necessário explicitamente instalar ou carregá-la.

Reviewer: ```suggestion
Você precisa da interface de linha de comando do Quarto (Quarto CLI -- *command line interface*), mas não é necessário instalá-la ou carregá-la explicitamente.
```

library(tidyverse)
```

## Básico de Quarto

Isso é um arquivo Quarto -- um arquivo de texto simples que possui a extensão `.qmd`:

```{r echo = FALSE, comment = ""}
cat(readr::read_file("quarto/diamond-sizes.qmd"))
```

O arquivo contém três tipos importantes de conteúdo:

1.  Um **cabeçalho YAML** (opcional) envolto por `---`.
2.  **Blocos** (ou *chunks*) de código R envoltos por ```` ``` ````.

> 3.  Texto misturado com formatação simples de texto como `# cabeçalho` e `_itálico_` (respectivamente, para denotar um título ou marcar texto como itálico).

Reviewer: ```suggestion
3.  Texto misturado com formatação simples de texto como `# título/subtítulo` e `_itálico_`.
```

@fig-diamond-sizes-notebook mostra um documento `.qmd` no RStudio com interface de caderno, no qual código e saída são intercalados.

> Você pode executar cada bloco de código clicando no ícone Executar -- *Run* (triângulo verde que parece com o botão de play no canto superior direito do bloco), ou pressionando Cmd/Ctrl + Shift + Enter.

Reviewer 2: ```suggestion
Você pode executar cada bloco de código clicando no ícone Executar -- *Run* (triângulo verde que parece com o botão de play no canto superior direito do bloco de código), ou pressionando Cmd/Ctrl + Shift + Enter.
```

> O RStudio executará o código e mostrará os resultados junto com o código.

Reviewer: ```suggestion
O RStudio executará o código e mostrará os resultados juntamente com o código.
```

```{r}
#| label: fig-diamond-sizes-notebook
#| echo: false
#| out-width: "90%"
#| fig-cap: |
#|   Um documento Quarto no RStudio. Código e saída intercalados 
#|   no documento, com a saída do gráfico aparecendo logo abaixo do código.
#| fig-alt: |
#|   Janela do RStudio com um documento Quarto intitulado "diamond-sizes.qmd" 
#|   à esquerda e um painel de Viewer em branco à direita. O documento 

> #|   Quarto tem um bloco de código que cria o gráfico de frequências dos diamantes 

Reviewer: ```suggestion
#|   Quarto tem um bloco de código que cria um gráfico de frequências dos diamantes 
```

#|   que pesam menos que 2.5 quilates. Esse gráfico mostra que a frequência 

> #|   diminui na medida que o peso aumenta.

Reviewer: ```suggestion
#|   diminui à medida que o peso aumenta.
```

knitr::include_graphics("quarto/diamond-sizes-notebook.png")
```

> Se você não gosta de ver seus gráficos e saída dos códigos no documento e prefere usar os painéis de console e gráfico (aba *plot*), é possível clicar no ícone de engrenagem  próximo ao botão "*Render*" e selecionar "*Chunk Output in Console*" (ou seja, mostrar resultados do chunk no console), conforme mostrado em @fig-diamond-sizes-console-output.

Reviewer: ```suggestion
Se você não gosta de ver seus gráficos e saída dos códigos no documento e prefere usar os painéis do console e gráfico (aba *plot*), é possível clicar no ícone de engrenagem  próximo ao botão "*Render*" e selecionar "*Chunk Output in Console*" (ou seja, mostrar resultados do chunk no console), conforme mostrado em @fig-diamond-sizes-console-output.
```

```{r}
#| label: fig-diamond-sizes-console-output
#| echo: false
#| out-width: "90%"
#| fig-cap: |
#|   Um documento Quarto no RStudio com a saída de gráfico no painel Plots.
#| fig-alt: |
#|   Janela do RStudio com um documento Quarto de nome "diamond-sizes.qmd" 
#|   à esquerda e o painel Plot no canto inferior direito. O documento

> #|   Quarto tem um bloco de código que cria o gráfico de frequências de diamantes

Reviewer: ```suggestion
#|   Quarto tem um bloco de código que cria um gráfico de frequências de diamantes
```

#|   pandoc e finalmente PDF, MS Word ou HTML.
#| fig-cap: |
#|   Diagrama do fluxo de trabalho Quarto de qmd a knitr, md, pandoc
#|   para saída nos formatos PDF, MS Word ou HTML.

knitr::include_graphics("images/quarto-flow.png")
```

Para iniciar seu próprio arquivo `.qmd`, selecione *File \> New File \> Quarto Document...* na barra de menu.
O RStudio abrirá uma janela em que você poderá preencher informações para preencher seu arquivo com conteúdos úteis que lembra como os principais recursos do Quarto funcionam.

As seções a seguir aprofundam os três componentes de um documento Quarto em mais detalhes: o texto em markdown, os blocos de código e o cabeçalho YAML.

### Exercícios

1.  Crie um novo documento Quarto usando *File \> New File \> Quarto Document*.
    Leia as instruções.
    Pratique a execução dos blocos de código individualmente.
    Depois, renderize o documento clicando no botão designado e depois usando o atalho do teclado.
    Verifique se consegue alterar o código, rodá-lo novamente e ver o novo documento modificado.

2.  Crie um novo documento Quarto para cada um dos três formatos disponibilizados: HTML, PDF e Word.
    Renderize cada um dos três documentos.

>     Qual é a diferença entre os resultados?

Reviewer: ```suggestion
    Qual é a diferença entre as saídas (*outputs*)?
```

>     Qual é a diferença entre os insumos?

Reviewer: ```suggestion
    Qual é a diferença entre as entradas (*inputs*)?
```

### Exercícios

1.  Crie novamente o documento em @fig-visual-editor usando o editor visual.
2.  Usando o editor visual, insira um bloco de código usando o menu Insert e depois faça o mesmo usando qualquer ferramenta.
3.  Usando o editor visual, descubra como:
    a.  Adicionar uma nota de rodapé.
    b.  Adicionar uma linha horizontal.
    c.  Adicionar um bloco de citação.
4.  Usando o editor visual, vá em Insert \> Citation e insira a citação para o artigo cujo título é [Welcome to the Tidyverse](https://joss.theoj.org/papers/10.21105/joss.01686) usando o DOI (*digital object identifier*), que é [10.21105/joss.01686](https://doi.org/10.21105/joss.01686). Renderize o documento e observe como a referencia aparece no documento. Que alteração você consegue identificar no YAML do seu documento?

## Editor Source {#sec-source-editor}

Também é possível editar documentos Quarto usando o editor Source no RStudio, sem a assistência do editor visual.
Enquanto o editor Visual será familiar para aqueles com experiência em escrita com ferramentas como Google docs, o editor Source será familia para aqueles com experiência escrevendo scripts R ou documentos R Markdown.

> O editor Source também pode ser útil para correções (*debugging*) qualquer erro de sintaxe Quarto, já que muitas vezes é mais fácil achar esses erros em texto simples.

Reviewer 2: ```suggestion
O editor Source também pode ser útil para procurar erros (*debugging*) de sintaxe Quarto, já que muitas vezes é mais fácil achar esses erros em texto simples.
```

cat(readr::read_file("quarto/markdown.qmd"))
```

A melhor forma de aprender essas coisas é simplesmente tentar.
Isso levará alguns dias, mas em pouco tempo se tornam algo automático e não será necessário pensar tanto sobre elas.
Se você se esquecer, pode buscar uma boa folha de referência em *Help \> Markdown Quick Reference*.

### Exercícios

> 1.  Pratique o que você aprendeu criando um CV breve.

Reviewer 2: ```suggestion
1.  Pratique o que você aprendeu criando um currículo (CV) breve.
```

    O título deve ser o seu nome e você deve incluir cabeçalhos para pelo menos sua educação e trabalho.
    Cada uma das seções deve incluir uma lista em tópicos de trabalhos/títulos.
    Destaque o ano em negrito.

2.  Usando o editor fonte e a referência rápida para Markdown, descubra como:

    a.  Adicionar uma nota de rodapé.
    b.  Adicionar uma linha horizontal.
    c.  Adicionar um bloco de citação.

> 3.  Copie e cole o conteúdo de `diamond-sizes.qmd` a partir de <https://github.com/hadley/r4ds/tree/main/quarto> para um arquivo R Quarto local.

Reviewer 2: ```suggestion
3.  Copie e cole o conteúdo de `diamond-sizes.qmd` a partir de <https://github.com/cienciadedatos/pt-r4ds/tree/main/quarto> para um arquivo R Quarto local.
```

    Verifique se você consegue executá-lo, depois adicione texto depois do polígono de frequências que descreve suas características mais chamativas.

> 4.  Crie um documento em Google doc ou MS Word (ou localize um documento que você tenha criado previamente) que contenha títulos, hiperlinks, texto formatado, etc.

Reviewer 2: ```suggestion
4.  Crie um documento em Google doc ou MS Word (ou use um documento que você tenha criado previamente) que contenha títulos, hiperlinks, texto formatado, etc.
```

    Copie os conteúdos desse documento e cole-os em um documento quarto no editor visual.

>     Em seguida, troque para o editor source e inspecion o código fonte.

Reviewer 2: ```suggestion
    Em seguida, troque para o editor source e inspecione o código fonte.
```

## Blocos de Código

Para executar código em um documento Quarto, é necessário incluir um bloco de código.
Há três maneiras de fazer isso:

1.  O atalho de teclado Cmd + Option + I / Ctrl + Alt + I.

2.  O botão "Insert" na barra de ferramentas do editor.

3.  Escrevendo manualmente os limitadores de bloco ```` ```{r} ```` e ```` ``` ````.

> Recomandamos que você aprenda o atalho de teclado.

Reviewer 2: ```suggestion
Recomendamos que você aprenda o atalho de teclado.
```

Isso economizará muito tempo no longo prazo!

É possível prosseguir para a execução do código usando o atalho que agora (temos fé!) que você conhece e ama:Cmd/Ctrl + Enter.

> No entanto, blocos recebem um novo atalho: Cmd/Ctrl + Shift + Enter, que executa todo o código no bloco.

Reviewer 2: ```suggestion
No entanto, blocos de código recebem um novo atalho: Cmd/Ctrl + Shift + Enter, que executa todo o código no bloco.
```

Pense no bloco de código como uma função.

> Um bloco deve ser relativamente autocontido, objetivando uma única tarefa.

Reviewer 2: ```suggestion
Um bloco deve ser relativamente autocontido, focado em realizar uma única tarefa.
```

> As seções a seguir descrevem o cabeçalho do bloco, que consiste em ```` ```{r} ````, seguido de um rótulo opcional e diversas outras opções de bloco, cada uma em sua própria linha, marcada por `#|`.

Reviewer 2: ```suggestion
As seções a seguir descrevem o cabeçalho do bloco de código, que consiste em ```` ```{r} ````, seguido de um rótulo opcional e diversas outras opções de bloco, cada uma em sua própria linha, marcada por `#|`.
```

> ### Rótulo de bloco

Reviewer 2: ```suggestion
### Rótulo do bloco de código
```

> Blocos podem receber um rótulo opcional, por exemplo

Reviewer 2: ```suggestion
Blocos de código podem receber um rótulo opcional, por exemplo
```

```{r}
#| echo: fenced
#| label: adicao-simples

1 + 1
```

Há três vantagens nisso:

> 1.  É possível navegar mais facilmente para blocos específicos usando o navegador em lista no canto inferior esquerdo do editor do script:

Reviewer 2: ```suggestion
1.  É possível navegar mais facilmente para blocos de código específicos usando o navegador em lista no canto inferior esquerdo do editor do script:
```

```{r}
#| echo: false
#| out-width: "30%"
#| fig-alt: |

> #|   trecho da IDE RStudio mostrando apenas o navegador de código drop-down

Reviewer 2: ```suggestion
#|   Trecho da IDE RStudio mostrando apenas o navegador de código drop-down
```

> #|   que msotra três blocos. Bloco 1 é setup. Bloco 2 é cars e

Reviewer 2: ```suggestion
#|   que mostra três blocos. Bloco 1 é setup. Bloco 2 é cars e
```

#|   nele há uma seção chamada Quarto. Bloco 3 é pressão (pressure) e 
#|   está em uma seção chamada Including plots.

knitr::include_graphics("screenshots/quarto-chunk-nav.png")
```

> 2.  Gráficos produzidos pelos blocos terão nomes úteis para reutilizá-los em outros lugares.

Reviewer 2: ```suggestion
2.  Gráficos produzidos pelos blocos de código terão nomes úteis para reutilizá-los em outros lugares.
```

    Mais sobre isso em @sec-figures.

> 3.  É possível organizar redes de bloco em cache para evitar múltiplas execuções de computações caras em toda execução.

Reviewer 2: ```suggestion
3.  É possível organizar redes de bloco de código em cache para evitar múltiplas execuções de computações caras em toda execução.
```

    Mais sobre isso em @sec-caching.

> Os rótulos dos seus blocos devem ser curtos, mas evocativos e não devem conter espaços.

Reviewer 2: ```suggestion
Os rótulos dos seus blocos de código devem ser curtos, mas evocativos e não devem conter espaços.
```

Recomendamos o uso de traços (`-`) para separar palavras (em vez de sublinhados, `_`) e evitar o uso de outros caracteres especiais nos rótulos de código.

> Em geral você é livre para rotular o bloco de código como quiser, mas há um bloco cujo nome indica um coportamento especial: `setup`.

Reviewer 2: ```suggestion
Em geral, você é livre para rotular o bloco de código como quiser, mas há um bloco cujo nome indica um coportamento especial: `setup`.
```

> Quando você está no modo caderno, o bloco com nome *setup* será executado automaticamente uma vez antes que todo o seu código seja executado.

Reviewer 2: ```suggestion
Quando você está no modo *notebook*, o bloco com nome *setup* será executado automaticamente uma vez antes que todo o seu código seja executado.
```

> Além disso,rótulos não podem ser duplicados.

Reviewer 2: ```suggestion
Além disso, rótulos não podem ser duplicados.
```

> Cada rótulo de bloco deve ser único.

Reviewer 2: ```suggestion
Cada rótulo de bloco de código deve ser único.
```

> ### Opções de bloco

Reviewer 2: ```suggestion
### Opções de bloco de código (*chunk options*)
```

> ### Opções de bloco de código (*chunk options*)

Author: Mas aparece na linha 56 :)

> ### Opções de bloco de código (*chunk options*)

Reviewer: ah nao lembrava, hehe pq demorei mil anos entre o dia que comecei e terminei hehe

> Saídas de bloco podem ser personalizadas com campos de opções (**options**) fornecidos para o cabeçalho do bloco.

Reviewer 2: ```suggestion
Saídas de blocos de código podem ser personalizadas com campos de opções (**options**) fornecidos para o cabeçalho do bloco.
```

Quase 60 opções para personalização dos seus blocos de código são fornecidas pelo Knitr.
Aqui cobriremos as opções mais importantes, que você utilizará com frequência.
É possível verificar a lista completa em [https://yihui.org/knitr/options](https://yihui.org/knitr/options/){.uri}.

> O conjunto mais importante de opçòes controla se seu código será executado e que resultados serão exibidos no relatório final:

Reviewer 2: ```suggestion
O conjunto mais importante de opções controla se seu código será executado e que resultados serão exibidos no relatório final:
```

-   `eval: false` previne o código de ser avaliado.
    (E obviamente, se o código não é executado, nenhum resultado será gerado).
    Isso é útil para exibir um código de exemplo ou para desabilitar um grande bloco de código sem comentar cada uma de suas linhas.

-   `include: false` executa o código, mas não o mostra nem inclui seus resultados no documento final.
    Use isso para códigos de configurações cujos resultados você não quer que dêem uma aparência de desordem no seu relatório.

-   `echo: false` evita o código mas permite a exibição dos resultados no arquivo finalizado.
    Use isso quando estiver escrevendo relatórios direcionados a pessoas que não querem ver o código por trás dos resultados.

-   `message: false` ou `warning: false` previne o aparecimento de mensagens ou avisos no arquivo final.

> -   `results: hide` esconde output; `fig-show: hide` esconde gráficos.

Reviewer 2: ```suggestion
-   `results: hide` esconde os resultados; `fig-show: hide` esconde gráficos.
```

> -   `error: true` permite a renderização contínua do seu documento mesmo que ocorra um erro.

Reviewer 2: ```suggestion
-   `error: true` permite que a renderização do seu documento continue mesmo que ocorra um erro.
```

    Você raramente precisará incluir isso na versão final do seu relatório, mas pode ser útil se você precisa corrigir um erro específico dentro do seu `.qmd`.
    Também é útil se você está ensinando R e quer incluir um erro deliberadamente.
    O padrão, `error: false` impede a renderização caso exista qualquer erro no documento.

> Cada uma dessas opções de bloco é adicionada ao cabeçalho do bloco, seguindo `#|`, por exemplo, no bloco a seguir o resultado não é impresso, já que `eval` está marcado como *false*.

Reviewer 2: ```suggestion
Cada uma dessas opções de bloco de código é adicionada ao cabeçalho do bloco, seguindo `#|`, por exemplo, no bloco a seguir o resultado não é impresso, já que `eval` está marcado como *false*.
```

```{r}
#| echo: fenced
#| label: multiplicacao-simples
#| eval: false

2 * 2
```

A tabela a seguir resume que tipos de output cada opção suprime:

> | Opção            | Executa código | Mostra código | Output | Gráficos | Mensagens | Avisos |

Reviewer 2: ```suggestion
| Opção            | Executa código | Mostra código | Resultados | Gráficos | Mensagens | Avisos |
```

### Código em linha

Existe outra maneira de inserir código R em um documento Quarto: diretamente no texto com `r inline()`.
Isso pode ser muito útil se você for citar propriedades dos seus dados no texto.
Por exemplo, o documento modelo usado no começo do capítulo tinha:

> Temos dados sobre `r inline('nrow(diamonds)')` diamantes.
> Apenas `r inline('nrow(diamonds) - nrow(smaller)')` são maiores que 2.5 quilates.
> A distribuição dos restantes é exibida abaixo:

Quando o relatório é renderizado, os resultados dessas computações são inseridas no texto:

> Temos dados sobre 53940 diamantes.
> Apenas 126 são maiores que 2.5 quilates.
> A distribuição dos restantes é exibida abaixo:

Ao inserir números no texto, `format()` é seu amigo.
A função permite configurar o número de `digits` (casas decimais) para que você não imprima o número com uma precisão ridícula e `big.mark` para facilitar a leitura dos números.
Você pode combiná-las em uma função auxiliar:

```{r}
comma <- function(x) format(x, digits = 2, big.mark = ",")
comma(3452345)
comma(.12358124331)
```

### Exercícios

> 1.  Adicione uma seção que explore como o tamanho de diamantes varia de acordo com corte, cor e claridade.

Author: transparência?

> 1.  Adicione uma seção que explore como o tamanho de diamantes varia de acordo com corte, cor e claridade.

Reviewer: Acho que é melhor. Ainda mais porque acabei de ver que na base de dados traduzida, do pacote dados, essa variavel ficou como "transparência". Acho que no livro estao mantendo a consistência com essa base traduzida.

> 1.  Adicione uma seção que explore como o tamanho de diamantes varia de acordo com corte, cor e claridade.

Author: ```suggestion
1.  Adicione uma seção que explore como o tamanho de diamantes varia de acordo com corte, cor e transparência.
```

#| fig-width: 4
#| out-width: "50%"
#| fig-alt: |

> #|   Gráfico de dispersão de milhagem de estrada vs. deslocamento dos carros, em que os pontos

Reviewer: ```suggestion
#|   Gráfico de dispersão de milhagem em rodovia vs. deslocamento dos carros, em que os pontos
```

#| fig-width: 6
#| out-width: "50%"
#| fig-alt: |

> #|   Gráfico de dispersão de milhagem de estrada vs. deslocamento dos carros, em que os pontos

Reviewer: ```suggestion
#|   Gráfico de dispersão de milhagem em rodovia vs. deslocamento dos carros, em que os pontos
```

não, sei acho rodovia uma tradução melhor pra highway... mas de novo, só uma sugestão

> #|   têm tamanho menor que o gráfico anterior e o texto dos eixos e rótulos têm tamanho

Reviewer: ```suggestion
#|   têm tamanho menor que os do gráfico anterior e o texto dos eixos e rótulos têm tamanho
```

#| fig-width: 8
#| out-width: "50%"
#| fig-alt: |

> #|   Gráfico de dispersão de milhagem de estrada vs. deslocamento dos carros, em que os pontos

Reviewer: ```suggestion
#|   Gráfico de dispersão de milhagem em rodovia vs. deslocamento dos carros, em que os pontos
```

> #|   têm tamanho ainda menor que o gráfico anterior e o texto dos eixos e rótulos têm tamanho

Reviewer: ```suggestion
#|   têm tamanho ainda menor que os do gráfico anterior e o texto dos eixos e rótulos têm tamanho
```

#|   ainda menor que o texto em volta.

plot
```

> Se você quiser garantir que o tamanho da fonte é consistente em todas as suas figuras, será necessário ajustar sempre o `out-width` com o `fig-width` para manter a mesma razão com o `out-width` padrão.

Reviewer: ```suggestion
Se você quiser garantir que o tamanho da fonte seja consistente em todas as suas figuras, será necessário ajustar sempre o `out-width` com o `fig-width` para manter a mesma razão com o `out-width` padrão.
```

Por exemplo, se seu `fig-width` padrão é 6 e o `out-width` está em "70%", será necessário configurar `fig-width` a 4.3 (6 \* 0.5 / 0.7) quando ajustar `out-width: "50%"`.

Entender dimensionamento é uma arte e uma ciência e acertar essas coisas pode requerer uma abordagem de tentativa e erro. Você pode aprender mais sobre isso no [post de blog tomando controle de dimensionamento de gráficos](https://www.tidyverse.org/blog/2020/08/taking-control-of-plot-scaling/).

### Outras opções importantes

> Quando estamos trabalhando com código e texto, como neste livro, é possível ajustar `fig-show: hold` de modo que os gráficos são exibidos após o código.

Reviewer: ```suggestion
Quando estamos trabalhando com código e texto, como neste livro, é possível ajustar `fig-show: hold` de modo que os gráficos sejam exibidos após o código.
```

Isso rende o efeito agradável de nos forçar a quebrar blocos de código grandes em menores com suas explicações.

Para adicionar uma legenda ao gráfico, use `fig-cap`.
No Quarto isso mudará a figura de *inline* (no meio do texto) para "*floating*".

> Se você está produzindo PDF, o tipo gráfico padrão é PDF.

Reviewer: ```suggestion
Se você está produzindo PDF, o tipo de gráfico padrão é PDF.
```

> Isso é um bom padrão porque PDF são gráficos vetoriais de alta qualidade.

Reviewer: ```suggestion
Isso é um bom padrão porque PDFs são gráficos vetoriais de alta qualidade.
```

Author: Tem plural de sigla assim?

Reviewer: tem no original em ingles.. hehe então só reproduzi hehe

No entanto, isso pode gerar gráficos muito grandes e lentos se você está exibindo milhares de pontos.

> Nesse caso, ajusete `fig-format: "png"` para forçar o uso de PNGs.

Reviewer: ```suggestion
Nesse caso, ajuste `fig-format: "png"` para forçar o uso de PNGs.
```

Nesta seção focaremos na segunda, tabelas geradas via computação.

Por padrão, Quarto imprime *data frames* e matrizes como você as veria no console:

```{r}
mtcars[1:5, ]
```

Se voce preferir que os dados sejam expostos com formatação adicional, você pode usar a função `knitr::kable()`.
O código abaixo gera @tbl-kable.

```{r}
#| label: tbl-kable
#| tbl-cap: Uma tabela feita com knitr (*kable*).

knitr::kable(mtcars[1:5, ], )
```

Leia a documentação para `?knitr::kable` para ver de que outras formas você pode personalizar a sua tabela.
Para uma personalização ainda mais profunda, considere os pacotes **gt**, **huxtable**, **reactable**, **kableExtra**, **xtable**, **stargazer**, **pander**, **tables**, e **ascii**.
Cada um fornece um conjunto de ferramentas para retornar tabelas formatadas a partir de código R.

### Exercícios

1.  Abra `diamond-sizes.qmd` no editor visual, insira um bloco de código e adicione uma tabela com `knitr::kable()` que mostre as 5 primeiras linhas do *data frame* `diamonds`.
2.  Exiba a mesma tabela com `gt::gt()`.

> 3.  Adicione um rótulo de bloco de código que começa com o prefixo `tbl-` e adicione a legenda para a tabela com a opção de bloco `tbl-cap`. Depois edite o texto acima do bloco de código para adicionar a referência cruzada para a tabela com Insert \> Cross Reference.

Reviewer: ```suggestion
3.  Adicione um rótulo de bloco de código que comece com o prefixo `tbl-` e adicione a legenda para a tabela com a opção de bloco `tbl-cap`. Depois edite o texto acima do bloco de código para adicionar a referência cruzada para a tabela com Insert \> Cross Reference.
```

## Salvando em cache {#sec-caching}

> Normalmente, cada renderização do documento começa com uma folha em branco.

Reviewer: ```suggestion
Normalmente, cada renderização do documento começa de uma folha em branco.
```
Achei que do outro jeito paecia que ia aparecer literalmente uma folha em branco no documento, mas acho que é pra dizer que começa "do zero"

Documentos HTML tipicamente têm uma série de dependências externas (por exemplo imagens, guias de estilo CSS, JavaScript, etc.) e, por padrão, o Quarto coloca essas dependências em uma pasta `_files` no mesmo diretório que o seu arquivo `.qmd`.
Se você publicar o arquivo HTML em uma plataforma (por exemplo, QuartoPub, <https://quartopub.com/>), as dependências desse diretório são publicadas com o documento e então estarão disponíveis no relatório publicado.
No entanto, se você quer enviar o relatório para um colega por email, é possível que prefira ter um único documento HTML autocontido que tem suas dependências todas embutidas.
Você pode fazer isso especificando a opção `embed-resources`.

``` yaml
format:
  html:
    embed-resources: true
```

O arquivo resultante será autocontido, de modo que não necessitará de qualquer arquivo interno nem de acesso à internet para ser exibido corretamente por um navegador.

### Parâmetros

Documentos Quarto podem incluir um ou mais parâmetros cujos valores podem ser ajustados quando se renderiza o relatório.
Parâmetros são úteis quando se quer executar mais vezes o mesmo relatório com valores diferentes para vários inputs chave.

> Por exemplo, pode ser que seja necessário produzir relatórios de vendas por escritório, resultados de provas por aluno ou mesmo resumos demográficos por país.

Reviewer: ```suggestion
Por exemplo, pode ser que seja necessário produzir relatórios de vendas por filial, resultados de provas por aluno ou mesmo resumos demográficos por país.
```

```

### Bibliografias e citações

Bibliografia e citações podem ser automaticamente geradas pelo Quarto em uma série de estilos.
A forma mais direta de adicionar citações e bibliografias no seu documento Quarto é usando o editor visual no RStudio.

Para adicionar a citação no editor visual, vá em Insert \> Citation.
Citações podem ser inseridas a partir de uma variedade de fontes:

1.  Referências [DOI](https://quarto.org/docs/visual-editor/technical.html#citations-from-dois) (*Digital Object Identifier*).

2.  Bibliotecas particulares ou de grupo do [Zotero](https://quarto.org/docs/visual-editor/technical.html#citations-from-zotero).

3.  Pesquisas na [Crossref](https://www.crossref.org/), [DataCite](https://datacite.org/), ou na [PubMed](https://pubmed.ncbi.nlm.nih.gov/).

4.  Seu documento de bibliografia (um arquivo `.bib` no diretório do seu documento).

Internamente, o editor visual usa a representação padrão para citações do markdown Pandoc (por exemplo, `[@citation]`).

Se você inserir uma citação usando um dos primeiros três métodos, o editor visual automaticamente criará um arquivo `bibliography.bib` para você e adicionará a sua referência a ele.

> Isso também adicionará um campo `bibliography` ao YAML do documento Na medida em que você insere mais referências, esse arquivo é povoado com as citações.

Reviewer: ```suggestion
Isso também adicionará um campo `bibliography` ao YAML do documento.
À medida em que você insere mais referências, esse arquivo é povoado com as citações.
```
nas acho que "Na medida em que" não tem esse sentido de "As" mas de causa e efeito ,

Também é possivel editar diretamente esse arquivo usando muitos formatos comuns como BibLaTeX, BibTeX, EndNote, Medline.

> Para criar a citação dentro do seu arquivo `.qmd` no editor fonte, use a chave composta de '\@' + identificador da citação do seu arquivo de bibliografia.

Reviewer: ```suggestion
Para criar a citação dentro do seu arquivo `.qmd` no editor source, use a chave composta de '\@' + identificador da citação do seu arquivo de bibliografia.
```

Lá em cima tava sendo chamado de source mesmo eu acho, não?

Em seguida, coloque a citação entre colchetes.
Seguem alguns exemplos:

``` markdown
Separe múltiplas citações com `;`: Blah blah [@smith04; @doe99].

É possível inserir comentários arbitrários nos colchetes: 
Blah blah [veja @doe99, pp. 33-35; também @smith04, cap. 1].

Remova os colchetes para criar uma citação direta: @smith04 diz blah, ou @smith04 [p. 33] diz blah.

Adicione um `-` antes da citação para suprimir o nome do autor:
Smith diz blah [-@smith04].
```

Quando o Quarto renderiza o seu arquivo, a bibliografia será construída e adicionada ao final do seu documento.
A bibliografia conterá cada uma das citações do seu arquivo de bibliografia, mas não é criado um título no documento para ela.
Como resultado, é uma prática comum finalizar o seu documento com um título de seção para a bibliografia como `# Referências` ou `# Bibliografia`.

É possível alterar o estilo das suas citações e bibliografia fazendo referência a um arquivo CSL (*citation style language* -- ou linguagem de estilo de citação) no campo `csl`:

``` yaml
bibliography: rmarkdown.bib
csl: apa.csl
```

> As with the bibliography field, your csl file should contain a path to the file.

Reviewer: ```suggestion
Assim como no campo de bibliografia, seu arquivo csl deve conter um caminho para o arquivo.
```
Achei estranho em ingles, acho que ele queria falar cls field e nao csl file

Você pode trabalhar interativamente dentro de um bloco, editando e executando novamente com Cmd/Ctrl + Shift + Enter.
Quando o trabalho estiver bom, você pode continuar e criar um novo bloco de código.

O Quarto também é importante porque integra estreitamente texto e código.
Isso faz do Quarto um ótimo **caderno de análise** (*analysis notebook*), porque permite desenvolver código e registrar seus pensamentos.
Um caderno de análise compartilha muitos dos mesmos objetivos de um caderno de laboratório clássico nas ciências exatas.
O caderno:

-   Permite gravar o que você fez e por que você fez.
    Independentemente de quão boa é a sua memória, se você não registrar o que fez, chegará um momento em que você esquecerá os detalhes importantes.
    Escreva para não se esquecer!

-   Apoia o pensamento rigoroso.
    É mais provável que você desenvolva uma análise robusta se você registrar seus pensamentos enquanto trabalha, depois pode continuar a refletir sobre eles.
    Isso também economiza tempo quando você eventualmente organizar sua análise para compartilhar com outras pessoas.

-   Ajuda outras pessoas a entenderem o seu trabalho.

>     É raro fazer análise de dados sozinho e você frequentemente fará parte de um time.

Reviewer: ```suggestion
    É raro fazer análise de dados sozinho e você frequentemente fará parte de uma equipe.
```
Não sei, achei que equipe traduzia melhor a ideia do que time

-   Se você passa muito tempo em uma ideia de análise e no final parece que não vai dar em nada, não apague!
    Faça uma breve anotação sobre o motivo de a análise ter falhado e deixe no caderno.
    Isso ajudará você a não cair na mesma rota quando voltar à análise no futuro.

-   Geralmente é melhor registrar dados fora do R.
    Se você precisa registrar um pequeno trecho dos dados, faça isso de forma explícita usando `tibble::tribble()`.

-   Se você descobrir um erro no arquivo de dados, nunca o modifique diretamente.
    Em vez disso, escreva código para corrigir o valor.
    Escreva o motivo do reparo.

-   Antes de finalizar os trabalhos do dia, garanta que é possível renderizar o caderno.
    Se está usando *caching*, limpe os caches.
    Isso permitirá a correção de problemas enquanto o código ainda está fresco na memória.

-   Se você quiser que seu código seja reprodutível a longo prazo (por exemplo, para voltar a trabalhar nele no mês ou ano seguinte), pode ser necessário registrar as versões dos pacotes que seu código usa.
    Uma abordagem rigorosa é usar **renv**, <https://rstudio.github.io/renv/index.html>, que guarda os pacotes no seu diretório do projeto.

>     Um *hack* rápido e sujo é incluir um bloco que execute `sessionInfo()` --- isso não permite facilmente recriar seus pacotes em seu estado atual, mas pelo menos ajuda a saber em que estado estavam.

Reviewer: pencei em traduzir para "macete" , acho um hack rapido e sujo muito estranho hahaha

-   Você criará muitos, muitos cadernos de análise no decorrer da sua carreira.
    Como você os organizará de modo que consiga achá-los no futuro?
    Recomendamos guardá-los em projetos individuais e criar um bom esquema de nomes.

## Resumo

> Neste capítulo apresentamos o Quarto para escrita e publicação reprodutível de documentos computacionais que incluem seu código e seu texto no mesmo lugar.

Reviewer: ```suggestion
Neste capítulo apresentamos o Quarto para escrita e publicação reprodutível de documentos computacionais que incluam seu código e seu texto no mesmo lugar.
```

> Você aprendeu a escrever em documentos Quarto no RStudio com o editor fonte ou visual, como blocos de código funcionam e como personalizar opções para cache de computações.

Reviewer: ```suggestion
Você aprendeu sobre escrever em documentos Quarto no RStudio com o editor fonte ou visual, como blocos de código funcionam e como personalizar opções para cache de computações.
```

> Além disso, você aprendeu sobre ajustes no cabeçalho YAML para criar documentos autocontidos ou parametrizados, assim como sobre citações e bibliografia.

Reviewer: ```suggestion
Além disso, você aprendeu sobre ajustes no cabeçalho YAML para criar documentos autocontidos ou parametrizados, assim como incluir citações e bibliografia.
```

> Também demos dicas de fluxo de trabalho e resolução de problemas.

Reviewer: ```suggestion
Também demos dicas de fluxo de trabalho e de resolução de problemas.
```

Enquanto essa introdução deve ser suficiente para iniciar seu trabalho com Quarto, ainda há muito a aprender.
Quarto ainda é relativamente jovem e ainda está crescendo rapidamente.
O melhor lugar para se atualizar das últimas inovações é o site oficial: [https://quarto.org](https://quarto.org/){.uri}.

Há dois tópicos importantes que não cobrimos aqui: colaboração e detalhes para comunicar sua ideia com precisão para outros humanos.

> Colaboração é parte vital da ciência de dados moderna e você pode facilitar muito a vida usando ferramentas de controle de versão, como Git e GitHub.

Reviewer: ```suggestion
Colaboração é uma parte vital da ciência de dados moderna e você pode facilitar muito a sua vida usando ferramentas de controle de versão, como Git e GitHub.
```

> Recomendamos o livro "Happy Git with R", uma introdução amigável a Git e GitHub para quem programa em R, escrito pela Jenny Bryan.

Reviewer: ```suggestion
Recomendamos o livro "Happy Git with R", uma introdução amigável ao Git e ao GitHub para quem programa em R, escrito pela Jenny Bryan.
```

Here we assume that the csl file is in the same directory as the .qmd file.
A good place to find CSL style files for common bibliography styles is <https://github.com/citation-style-language/styles>.

## Fluxo de trabalho

> Anteriormente discutimos um fluxo de trabalho básico para capturar seu código R quando é trabalhado interativamente no *console*, depois capturar o que funciona no *editor de script*.

Author: @cmusso86 quando for revisar, dá uma olhada nesse pedaço!

Quarto ainda é relativamente jovem e ainda está crescendo rapidamente.
O melhor lugar para se atualizar das últimas inovações é o site oficial: [https://quarto.org](https://quarto.org/){.uri}.

Há dois tópicos importantes que não cobrimos aqui: colaboração e detalhes para comunicar sua ideia com precisão para outros humanos.
Colaboração é parte vital da ciência de dados moderna e você pode facilitar muito a vida usando ferramentas de controle de versão, como Git e Github.
Recomendamos "Happy Git with R", uma introdução amigável a Git e Github de usuários de R, por Jenny Bryan.
O livro está disponível online com acesso livre: <https://happygitwithr.com>.

Também não comentamos sobre o que você deveria realmente escrever para comunicar seus resultados da sua análise com clareza.
Para melhorar sua escrita, recomendamos muito a leitura de [*Style: Lessons in Clarity and Grace*](https://www.amazon.com/Style-Lessons-Clarity-Grace-12th/dp/0134080416) por Joseph M. Williams & Joseph Bizup, ou [*The Sense of Structure: Writing from the Reader's Perspective*](https://www.amazon.com/Sense-Structure-Writing-Readers-Perspective/dp/0205296327) by George Gopen.

> Ambos os livros ajudarão a entender a estrutura de frases e parágrafos e darão gerramentas para tornar sua escrita mais clara.

Reviewer 2: ```suggestion
Ambos os livros ajudarão a entender a estrutura de frases e parágrafos e darão ferramentas para tornar sua escrita mais clara.
```
